### PR TITLE
Decompiled four non-matching functions.

### PR DESCRIPTION
**Pull request by AntonioCastelli**
Had these laying around for a while. Figured I should push them.

### CODE REVIEW

File: src/unknown_00BC20.c [hunk 1]
```diff
@@ -600,7 +600,72 @@ void func_8000E1EC(Object *object, s32 arg1) {
...
+    *D_8011AEEC = (s32) player;
+    *D_8011AEE8 = (s32) player;
+    player_64 = player->unk64;
+    player_64->unk1D6 = (s8) D_8011AD45;
+    player_64->unk0_a.unk0_b.unk2 = (u8)0;
+    player_64->unk1D7 = (s8) D_8011AD45;
+    player_64->unk0_a.unk0_b.unk0 = (u16)0;
+    player_64->unk118 = 0;
+    player_64->unk0_a.unk0_b.unk3 = (s8) settings->racers[0].character;
+    if ((get_filtered_cheats() & 0x10) != 0) {
```
Comment (Faschz) [inline src/unknown_00BC20.c:656]:
```suggestion
    if (get_filtered_cheats() & CHEAT_BIG_CHARACTERS) {
```

Cheats have defines in menu.h

File: src/unknown_00BC20.c [hunk 2]
```diff
@@ -600,7 +600,72 @@ void func_8000E1EC(Object *object, s32 arg1) {
...
+        player->scale *= 1.4f;
+    }
+    if ((get_filtered_cheats() & 0x20) != 0) {
```
Comment (Faschz) [inline src/unknown_00BC20.c:659]:
```suggestion
    if (get_filtered_cheats() & CHEAT_SMALL_CHARACTERS) {
```

Same as above

File: src/unknown_0348C0.c [hunk 3]
```diff
@@ -1184,7 +1184,60 @@ void obj_loop_world_key(Object *worldKeyObj, s32 speed) {
...
+    f32 temp_f0;
+    s32 temp_v0;
+    Object_64 *temp_v0_2;
+    f32 phi_f0;
+
+    arg0->unk4C->unk14 = 2;
+    arg0->unk4C->unk11 = 4;
+    arg0->unk4C->unk10 = 0x14;
+    temp_v0 = get_filtered_cheats();
+    if (temp_v0 & 0x20000) {
```
Comment (Faschz) [inline src/unknown_0348C0.c:1205]:
```suggestion
    if (temp_v0 & CHEAT_ALL_BALLOONS_ARE_BLUE) {
```

File: src/unknown_0348C0.c [hunk 4]
```diff
@@ -1184,7 +1184,60 @@ void obj_loop_world_key(Object *worldKeyObj, s32 speed) {
...
+        arg1->unk9 = 0U;
+    } else if (temp_v0 & 0x08000) {
```
Comment (Faschz) [inline src/unknown_0348C0.c:1207]:
```suggestion
    } else if (temp_v0 & CHEAT_ALL_BALLOONS_ARE_RED) {
```

File: src/unknown_0348C0.c [hunk 5]
```diff
@@ -1184,7 +1184,60 @@ void obj_loop_world_key(Object *worldKeyObj, s32 speed) {
...
+        arg1->unk9 = 1U;
+    } else if (temp_v0 & 0x10000) {
```
Comment (Faschz) [inline src/unknown_0348C0.c:1209]:
```suggestion
    } else if (temp_v0 & CHEAT_ALL_BALLOONS_ARE_GREEN) {
```

File: src/unknown_0348C0.c [hunk 6]
```diff
@@ -1184,7 +1184,60 @@ void obj_loop_world_key(Object *worldKeyObj, s32 speed) {
...
+    phi_f0 = arg1->unkA;
+    if (arg1->unkA < 10.0f) {
+        phi_f0 = 10.0f;
+    }
+    temp_f0 = phi_f0 / 64.0f;
+    arg0->scale = (f32) (arg0->descriptor_ptr->unkC * temp_f0);
+    arg0->unk64->unk4 = 0;
+    arg0->unk64->unk0_a.unk0 = arg0->scale;
+    arg0->unk7C.word = 0;
+    if ((get_filtered_cheats() & 0x800) != 0) {
```
Comment (Faschz) [inline src/unknown_0348C0.c:1233]:
```suggestion
    if (get_filtered_cheats() & CHEAT_DISABLE_WEAPONS) {
```

File: src/unknown_0348C0.c [hunk 7]
```diff
@@ -1184,7 +1184,60 @@ void obj_loop_world_key(Object *worldKeyObj, s32 speed) {
...
+    arg0->unk4C->unk11 = 4;
+    arg0->unk4C->unk10 = 0x14;
+    temp_v0 = get_filtered_cheats();
+    if (temp_v0 & 0x20000) {
+        arg1->unk9 = 0U;
+    } else if (temp_v0 & 0x08000) {
+        arg1->unk9 = 1U;
+    } else if (temp_v0 & 0x10000) {
+        arg1->unk9 = 2U;
+    } else if (temp_v0 & 0x40000) {
```
Comment (Faschz) [inline src/unknown_0348C0.c:1211]:
```suggestion
    } else if (temp_v0 & CHEAT_ALL_BALLOONS_ARE_YELLOW) {
```

File: src/unknown_0348C0.c [hunk 8]
```diff
@@ -1184,7 +1184,60 @@ void obj_loop_world_key(Object *worldKeyObj, s32 speed) {
...
+        arg1->unk9 = 3U;
+    } else if (temp_v0 & 0x80000) {
```
Comment (Faschz) [inline src/unknown_0348C0.c:1213]:
```suggestion
    } else if (temp_v0 & CHEAT_ALL_BALLOONS_ARE_RAINBOW) {
```

File: src/unknown_00BC20.c [hunk 9]
```diff
@@ -600,7 +600,72 @@ void func_8000E1EC(Object *object, s32 arg1) {
...
+    sp2C.common.z = D_8011AD4A;
+    sp2C.unkC = D_8011AD4C;
+    func_800521B8(1);
+    player = func_8000EA54(&sp2C, 0x11);
+    gObjectCount = 1;
+    (*gObjectStructArrayPtr)[0] = (s32) player;
+    *D_8011AEEC = (s32) player;
+    *D_8011AEE8 = (s32) player;
+    player_64 = player->unk64;
+    player_64->unk1D6 = (s8) D_8011AD45;
```
Comment (Ryan-Myers) [inline src/unknown_00BC20.c:650]: unk1D6 is undefined here. Check to see if maybe you forgot to update the Object_64 struct in this PR?

File: src/unknown_00BC20.c [hunk 10]
```diff
@@ -600,7 +600,72 @@ void func_8000E1EC(Object *object, s32 arg1) {
...
+    player_64->unk0_a.unk0_b.unk2 = (u8)0;
+    player_64->unk1D7 = (s8) D_8011AD45;
+    player_64->unk0_a.unk0_b.unk0 = (u16)0;
+    player_64->unk118 = 0;
```
Comment (Ryan-Myers) [inline src/unknown_00BC20.c:654]: unk118 is also undefined.

**Review by Ryan-Myers — CHANGES_REQUESTED** [review]
I'm good with these changes once we get the NON_MATCHING builds working again.

File: src/unknown_00BC20.c (hunk 9 above)
Comment (AntonioCastelli) [reply src/unknown_00BC20.c:651]: Done in https://github.com/DavidSM64/Diddy-Kong-Racing/pull/178/commits/e93a5db1ecfa2e29a151e673a3f295464e067349. Thanks.

**Review by Ryan-Myers — COMMENTED** [review]
Looks good!

**Pull request merged**